### PR TITLE
properties for $campaign_open for the cta url and if the button was p…

### PR DESCRIPTION
**Pull request by pchien**
…rimary or secondary. also now always track $campaign_open if they tap on either button or mini, instead of only if there was a ctal url

this is currently untested. i don't know how to use gradle to point the test app at my local changes. will ask in person later this week

### CODE REVIEW

**Comment by pchien** [comment]
okay i tested now. it seems to work!

File: src/main/java/com/mixpanel/android/mpmetrics/InAppFragment.java [hunk 1]
```diff
@@ -148,11 +152,18 @@ public boolean onSingleTapUp(MotionEvent event) {
...
                     }
+
+                    try {
+                        trackingProperties = new JSONObject();
+                        trackingProperties.put("url", uriString);
+                    } catch (final JSONException e) {
+                        MPLog.e(LOGTAG, "Can't put url into json properties");
+                    }
                 }
+                mMixpanel.getPeople().trackNotification("$campaign_open", inApp, trackingProperties);
```
Comment (patedit) [inline src/main/java/com/mixpanel/android/mpmetrics/InAppFragment.java]: do we still want to track campaign open even if the mini in-app does nothing?

File: src/main/java/com/mixpanel/android/takeoverinapp/TakeoverInAppActivity.java [hunk 2]
```diff
@@ -187,11 +191,31 @@ public void onClick(View v) {
...
+                    }
+                    try {
+                        if (trackingProperties == null) {
+                            trackingProperties = new JSONObject();
+                        }
+                        trackingProperties.put("button", whichButton);
+                    } catch (final JSONException e) {
+                        MPLog.e(LOGTAG, "Can't put button type into json properties");
                     }
+                    mMixpanel.getPeople().trackNotification("$campaign_open", inApp, trackingProperties);
```
Comment (patedit) [inline src/main/java/com/mixpanel/android/takeoverinapp/TakeoverInAppActivity.java]: same as before

File: src/main/java/com/mixpanel/android/mpmetrics/InAppFragment.java (hunk 1 above)
Comment (pchien) [reply src/main/java/com/mixpanel/android/mpmetrics/InAppFragment.java]: i think so. it tracks taps, same as normal inapps

**Pull request merged**